fix(catalog): missing type declarations for @maneki/charts

## Bug
`npx tsc --noEmit` in `apps/catalog` fails with:

```
src/pages/bar-chart.ts(3,38): error TS7016: Could not find a declaration file for module '@maneki/charts'
src/pages/line-chart.ts(3,39): error TS7016: Could not find a declaration file for module '@maneki/charts'
src/pages/scatter-chart.ts(3,42): error TS7016: Could not find a declaration file for module '@maneki/charts'
```

## Fix
Either add `tsc --emitDeclarationOnly` to the charts package build, or add a `declare module '@maneki/charts'` shim in the catalog.

--- a/packages/charts/moon.yml
+++ b/packages/charts/moon.yml
@@ -4,7 +4,7 @@ language: 'typescript'
 
 tasks:
   build:
-    script: 'tsc && vite build'
+    script: 'vite build && tsc --emitDeclarationOnly'
     inputs:
       - 'src/**/*'
       - 'tsconfig.json'

--- a/packages/charts/package.json
+++ b/packages/charts/package.json
@@ -11,7 +11,7 @@
   ],
   "scripts": {
     "dev": "vite",
-    "build": "tsc && vite build",
+    "build": "vite build && tsc --emitDeclarationOnly",
     "preview": "vite preview",
     "test": "vitest --run",
     "test:watch": "vitest"

